analysis/runtime/src/runtime/backend.rs

```diff
@@ -80,18 +83,32 @@ pub enum Backend {
 }
 
 impl Backend {
-    fn write_all(&mut self, events: impl IntoIterator<Item = Event>) {
-        for event in events {
+    fn write_all(&mut self, events: Arc<ArrayQueue<Event>>) {
+        let backoff = Backoff::new();
+        loop {
+            let event = match events.pop() {
+                Some(event) => event,
+                None => {
+                    // We can't use anything with a futex here since
+                    // the event sender might run inside a signal handler
+                    backoff.snooze();
```

Comment:
```suggestion
                    // We can't block on a lock/semaphore here since
                    // it might not be safe for the event sender to wake
                    // us from inside a signal handler
                    backoff.snooze();
```
Took me a minute to figure out how a futex here would relate to the event sender's signal handler